### PR TITLE
Update energy_system_component.rst

### PR DESCRIPTION
**Pull request by srhbrnds**
**Changes proposed in this pull request**:
-  deleted link to energy_type-def in Attribute name energy_type
-  fix some typos
-  proposed new Attribute names 
-  add comments and request changes to Requirements

Steps that need to be done before merge: 
- [x] clarify how to refer to the view-component and its attributes in the description e.g. *energy system component*, 'energy system component', Energy system component

### CODE REVIEW

File: docs/view_components/energy_system_component.rst [hunk 1]
```diff
@@ -8,17 +8,16 @@ Attributes
...
-:ref:`energy-type-def`
-
-    Energy type linked to the different energy sectors (e.g. heat, electricity, ...)
+**Energy type**
+    ref:`energy-type-def` linked to the different energy sectors (e.g. heat, electricity, ...)
 
 
 **Component type**
-    Can be one of (Source, sink, transformer, storage, bus)
+    Can be one of [Source, Sink, Transformer, Storage, Bus, ...] .. we propably not going to only use oemof.solph basic components in the tool, so list needs to be extended
```
Comment (Bachibouzouk) [inline docs/view_components/energy_system_component.rst]: I agree, however the ... implies for me that somewhere else the full list of components exists and I wonder then where is that list. I the list is closed it is clear that these are all values a component can take up to now (this list can however become bigger in the future). We can describe the component types more in detail somewhere else in RTD and link to this place after the open-ended list

File: docs/view_components/energy_system_component.rst [hunk 2]
```diff
@@ -8,17 +8,16 @@ Attributes
...
 
 
 **Unique id**
-    A positive integer number to identify the component uniquely within the list of energy system components of the :ref:`es_network-label`
+    A positive integer number to identify the component uniquely within the list of energy system components of the :ref:`es_network-label` .. why unique integer, rather than string label?
```
Comment (Bachibouzouk) [inline docs/view_components/energy_system_component.rst]: we can have a label as well, the integer id allows easily to garantee the uniqueness (for every new component we bump the number by one. These attributes are characterising the view-component but are not necessarily rendered (ie visible to enduser)

File: docs/view_components/energy_system_component.rst [hunk 3]
```diff
@@ -31,29 +30,31 @@ Attributes
...
     A list of other components's ids which are connected to this component and drain energy from it
 
     Requirements index: 1, 5, 6
 
 Actions
 ^^^^^^^
 
-1. Double-clicking on the rendering of the energy system component allow the user to edit its attributes (except id)
+1. Double-clicking on the rendering of the *energy system component* allow the user to edit its attributes (except id)
+2. As long as the *energy system component* is not connected correctly to another *energy system component* the Icon-frame is displayed in another colour
```
Comment (Bachibouzouk) [inline docs/view_components/energy_system_component.rst]: I would say this is a requirement, as there is nothing "actionnable" on the user side here

File: docs/view_components/energy_system_component.rst [hunk 4]
```diff
@@ -31,29 +30,31 @@ Attributes
...
 
 Requirements
 ^^^^^^^^^^^^
-1. An energy system component cannot de defined without being connected to another energy system component
-2. A component of type Source cannot be connected to a component of type Sink
-3. A component which is not of type bus, should have at least one connection to component of type bus bus
-4. A component which is of bus type, should have at least one source and one sink type connected to it
-5. A component of type Sink do not have Outward energy connection
-6. A component of type Source do not have Inward energy connection
+1. An *energy system component* cannot be defined without being connected to another *energy system component* .. what about the first component you define?
```
Comment (Bachibouzouk) [inline docs/view_components/energy_system_component.rst]: Good point, we can actually move your newly defined action 2. here

File: docs/view_components/energy_system_component.rst [hunk 5]
```diff
@@ -31,29 +30,31 @@ Attributes
...
+2. An *energy system component* which is not of *component type* Bus can not be connected to another *energy system component*, which is not of *component type* Bus
+3. An *energy system component* which is not of *component type* Bus, should have at least one connection to an *energy system component* of *component type* Bus
+4. An *energy system component* which is of *component type* Bus, should have at least one Source and one Sink *component type* connected to it .. I would delete this one, you can connect to buses with a Transformer, imagine you have an ac-electricity bus and an dc-electricity bus and you want to connect connect those two, you would e.g. connect via a Transfomer
```
Comment (Bachibouzouk) [inline docs/view_components/energy_system_component.rst]:
```suggestion
```
Agreed

File: docs/view_components/energy_system_component.rst [hunk 6]
```diff
@@ -31,29 +30,31 @@ Attributes
...
+5. An *energy system component* of *component type* Sink has a connection to *Input energy flow*, but no connection to *Output energy flow*
+6. An *energy system component* of *component type* Source has a connection *Output Energy flow*, but no connection to *Input energy flow*
+7. ... .. to indicate that there are more requirements to be added, if more component types are introduced
```
Comment (Bachibouzouk) [inline docs/view_components/energy_system_component.rst]: I would rather add more requirements then

**Review by Bachibouzouk — APPROVED** [review]
Thanks @srhbrnds , I approve now but let's agree or discuss further the point I raised before you merge :)

File: docs/view_components/energy_system_component.rst (hunk 1 above)
Comment (srhbrnds) [reply docs/view_components/energy_system_component.rst]: I agree, with this point. I opened issue #80 to continue this discussion.

File: docs/view_components/energy_system_component.rst (hunk 2 above)
Comment (srhbrnds) [reply docs/view_components/energy_system_component.rst]: okay, i'm fine with that.

**Pull request merged**